### PR TITLE
Add output for the file provider verification

### LINKED ISSUE #4793: Discussion: Verification command output
Filing this as a ticket because I don't see a straightforward implementation. Right now when we run a verification command and it fails all the user sees is something like this:

``````
   Error executing action `create` on resource 'monit_check[nginx]'
    ================================================================================

    Chef::Exceptions::ValidationFailed
    ----------------------------------
    file[/etc/monit/conf.d/nginx.conf] (/var/chef/cache/cookbooks/poise-monit/files/halite_gem/poise_monit/resources/monit_config.rb line 90) had an error: Chef::Exceptions::ValidationFailed: Proposed content for /etc/monit/conf.d/nginx.conf failed verification #<Chef::Resource::File::Verification:0x00000003dc3ae0>

    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/poise/files/halite_gem/poise/helpers/notifying_block.rb:69:in `notifying_block'
    /var/chef/cache/cookbooks/poise-monit/files/halite_gem/poise_monit/resources/monit_config.rb:72:in `action_create'```
``````

Notably we don't show the actual verification command if there was one (easy to fix, I can handle that) and we don't show the command output. Most tools with a syntax check option (`nginx -t`, `monit -t`) will print some debugging message indicating what the error is, which would be valuable to bubble up the chain. This is difficult though, because we are re-using the resource guard system and that doesn't expose command output as part of its API. A possible work-around is that good cookbooks shouldn't use the command form of `verify` and should instead use a block with `shell_out!` more directly, and raise the exception themselves. That's a bit weird though. What say we all.

### PR DESCRIPTION
**Pull request by lamont-granquist**
closes #4793

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@vsingh-msys or @Vasu1105 can one of you pick this up and fix the spec failures and add a spec or two for the behavior here?

**Comment by vsingh-msys** [comment]
Alright @lamont-granquist, will take it up and revert you back accordingly. Thanks

**Comment by lamont-granquist** [comment]
ping @vsingh-msys

File: lib/chef/provider/file.rb [hunk 1]
```diff
@@ -341,7 +341,10 @@ def do_validate_content
         if tempfile
           new_resource.verify.each do |v|
             unless v.verify(tempfile.path)
-              raise Chef::Exceptions::ValidationFailed.new "Proposed content for #{new_resource.path} failed verification #{new_resource.sensitive ? "[sensitive]" : v}"
+              backupfile = "#{Chef::Config[:file_cache_path]}/failed_validations/#{::File.basename(tempfile.path)}"
+              FileUtils.mkdir_p File.dirname(backupfile)
```
Comment (vsingh-msys) [inline lib/chef/provider/file.rb]: Getting error 
```
 NoMethodError
    -------------
    undefined method `dirname' for Chef::Provider::File:Class
```
Need to replace `File` with `::File`

**Comment by vsingh-msys** [comment]
@lamont-granquist sorry for the delayed response.

Started looking into it, will update as soon as possible. Thanks

**Comment by vsingh-msys** [comment]
Hi @lamont-granquist , Could you please have a look again. I have only fixed the specs and tested the expected behavior seems its working file. Also, let me know if any further queries/concerns.

Thanks

**Pull request merged**